maggy/experiment.py

```diff
@@ -104,16 +110,37 @@ def lagom(map_fun, searchspace=None, optimizer=None, direction='max', num_trials
 
         num_executors = util.num_executors()
 
-        if num_executors > num_trials:
-            num_executors = num_trials
+        # start experiment driver
+        if experiment_type == 'optimization':
+
+            assert num_trials > 0, "number of trials should be greater than zero"
+            if num_executors > num_trials:
+                num_executors = num_trials
+
+            exp_driver = ExperimentDriver('optimization', searchspace=searchspace, optimizer=optimizer,
+                                          direction=direction, num_trials=num_trials, name=name,
+                                          num_executors=num_executors, hb_interval=hb_interval, es_policy=es_policy,
+                                          es_interval=es_interval, es_min=es_min, description=description,
+                                          app_dir=app_dir, log_dir=log_dir, trial_dir=trial_dir)
```

Comment:
One additional indentation level is enough, arguments don't need to align with opening parenthesis of the function call. This also makes the lines too long (>79). But not tragic.
```suggestion
            exp_driver = ExperimentDriver('optimization', searchspace=searchspace,
                optimizer=optimizer, direction=direction, num_trials=num_trials,
                name=name, num_executors=num_executors, hb_interval=hb_interval,
                es_policy=es_policy, es_interval=es_interval, es_min=es_min,
                description=description, app_dir=app_dir, log_dir=log_dir,
                trial_dir=trial_dir)
```